my-dsl.md

```diff
@@ -2,37 +2,54 @@
+what happens when your DSL program runs._
 
+When a Flask program runs, it is interpreted as a Python program. The following explains how a complete web application built with Flask might execute, and also showcases the Model-View-Controller model:
+
+1. A user issues a request for the domain's root URL to go to the homepage. 
+
+2. Some URL routing mechanism will map the URL to a Python function.
+
+3. The Python function finds a web template which will fetch any images, CSS, etc.
+
+4. The rendered HTML is sent back to the URL routing mechanism, which sends it to the browser.
 
 # DSL-ness
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
+Flask is closer to the domain-specific end of the spectrum, as it is specific to web development. Because Flask is an intenral DSL, it contains all the programming logic of Python; however, because it was written and is used primarily for URL routing, HTML templating, and database manipulation, I think it is definitely considered more domain-specific (specific for web development) than general-purpose.
 
 # Internal or external?
 _Is the language implemented as an internal or external DSL? 
 Justify your answer._
 
+Flask would be considered an internal DSL, as it uses a general purpose language (Python) in a particular way (web development). 
 
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
+Flask is written in Python.
 
 # Benefits
 _Identify one potential benefit of the DSL: how is a programmer's life or a 
 company's bottom line made easier by the existence of this language?_
 
+By building a web application with Flask, developers take advantage of the lightweight framework, avoiding much of the overhead typically attributed to a more "full stack" web framework. Additionally, Flask provides more flexibility and granular control with its minimal framework, leaving design choices ultimately up to the developer. Flask also allows developers to take advantage of the Model-View-Controller model, which also contributes to more granular control. 
 
 # Drawbacks
 _Identify one potential drawback of the DSL: what does a programmer or company 
 lose by using this DSL instead of a general-purpose language?_
+
+One disadvantage of Flask is the lack of many features normally included in a "full stack" web framework, such as a data abstraction layer, form validation, etc. As such, for larger projects, web applications built with Flask require several additional libraries to fulfill these features -- several describe Flask as a "build what you want and bolt on whatever you need" framework, whereas other frameworks are described as an already complete package. However, it is worthwhile to note that not including these features make most of the aforementioned benefits possible (lightweight framework, granular control, flexibility).
```

Comment:
I'm a little confused on how having less features allows for more granular control... I do like how you compare Flask to other similar DSLs though.